lib/bundler/errors.rb

```diff
@@ -73,6 +73,27 @@ def message
     status_code(23)
   end
 
+  class TemporaryResourceError < BundlerError
```

Comment:
Ah thanks. I was having some trouble trying to DRY up that part with a private module method.

Is it an issue that this TemporaryResourceError would be inheriting from PermissionsError when it's not necessarily always something that's thrown from a lack of file permissions? Things like a temporary inability to fork or memory resource issues can cause this error as well.